### PR TITLE
Bug fixes

### PR DESCRIPTION
**Pull request by e5l**
- Add WebSocket config
- Fix coverage source sets
- Add iOS certificate pinning
- Add js origin support
- Multi-handle double close
- Prevent attributes freeze
- Fix native failed test report
- Fix default transformers headers
- Bug fixes

### CODE REVIEW

File: ktor-client/ktor-client-features/ktor-client-logging/api/ktor-client-logging.api [hunk 1]
```diff
@@ -31,13 +31,12 @@ public final class io/ktor/client/features/logging/LoggerKt {
...
 public final class io/ktor/client/features/logging/Logging {
 	public static final field Companion Lio/ktor/client/features/logging/Logging$Companion;
-	public fun <init> (Lio/ktor/client/features/logging/Logger;Lio/ktor/client/features/logging/LogLevel;)V
 	public fun <init> (Lio/ktor/client/features/logging/Logger;Lio/ktor/client/features/logging/LogLevel;Ljava/util/List;)V
+	public synthetic fun <init> (Lio/ktor/client/features/logging/Logger;Lio/ktor/client/features/logging/LogLevel;Ljava/util/List;ILkotlin/jvm/internal/DefaultConstructorMarker;)V
 	public final fun getFilters ()Ljava/util/List;
 	public final fun getLevel ()Lio/ktor/client/features/logging/LogLevel;
 	public final fun getLogger ()Lio/ktor/client/features/logging/Logger;
 	public final fun setFilters (Ljava/util/List;)V
-	public final fun setLevel (Lio/ktor/client/features/logging/LogLevel;)V
```
Comment (dmitrievanthony) [inline ktor-client/ktor-client-features/ktor-client-logging/api/ktor-client-logging.api:40]: Do we remove this method and the `init` above from the public API?

File: ktor-client/ktor-client-ios/darwin/src/io/ktor/client/engine/ios/certificates/CertificatesInfo.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+/**
+ * Certificate headers
+ *
+ * Sources for values:
+ * https://github.com/datatheorem/TrustKit/blob/master/TrustKit/Pinning/TSKSPKIHashCache.m
+ * https://github.com/IBM-Swift/BlueRSA/blob/master/Sources/CryptorRSA/CryptorRSAUtilities.swift
+ */
+internal object CertificatesInfo {
+    val rsa = mapOf(
```
Comment (dmitrievanthony) [inline ktor-client/ktor-client-ios/darwin/src/io/ktor/client/engine/ios/certificates/CertificatesInfo.kt:52]: Shouldn't we mark these values as `@SharedImmutable` also?

File: ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/JsonTest.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import io.ktor.client.features.json.*
+import io.ktor.client.features.json.serializer.*
+import io.ktor.client.request.*
+import io.ktor.client.tests.utils.*
+import kotlinx.serialization.*
+import kotlin.test.*
+
+class JsonTest : ClientLoader() {
+
```
Comment (dmitrievanthony) [inline ktor-client/ktor-client-tests/common/test/io/ktor/client/tests/JsonTest.kt:15]: Too many empty lines.

File: ktor-client/ktor-client-ios/darwin/src/io/ktor/client/engine/ios/certificates/CertificatesInfo.kt (hunk 2 above)
Comment (e5l) [reply ktor-client/ktor-client-ios/darwin/src/io/ktor/client/engine/ios/certificates/CertificatesInfo.kt:52]: Objects are shared by default

**Comment by petersamokhin** [comment]
Waiting for the CertificatePinner 🙂

**Pull request merged**